src/vm/jitinterface.cpp

```diff
@@ -4503,6 +4503,63 @@ CorInfoType CEEInfo::getTypeForPrimitiveValueClass(
     return result;
 }
 
+/*********************************************************************/
+CorInfoType CEEInfo::getTypeForPrimitiveNumericClass(
+        CORINFO_CLASS_HANDLE clsHnd)
+{
+    CONTRACTL {
+        SO_TOLERANT;
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    CorInfoType result = CORINFO_TYPE_UNDEF;
+
+    JIT_TO_EE_TRANSITION();
+
+    TypeHandle th(clsHnd);
+    _ASSERTE (!th.IsGenericVariable());
+
+    MethodTable    *pMT = th.GetMethodTable();
+    PREFIX_ASSUME(pMT != NULL);
+
+    // Is it a non primitive struct such as
+    // RuntimeTypeHandle, RuntimeMethodHandle, RuntimeArgHandle?
+    if (pMT->IsValueType() &&
+        !pMT->IsTruePrimitive()  &&
+        !pMT->IsEnum())
+    {
+        // default value CORINFO_TYPE_UNDEF is what we want
+    }
+    else
+    {
+        CorElementType ty = th.GetInternalCorElementType();
```

Comment:
You can rather call `GetSignatureCorElementType` here. Once you do that, you do not really need the IsValueType, IsTruePrimitive, IsEnum condition above.

Reply:
Got it, will change.